build_scripts/api/dreambooth/utils/model_utils.py

```
@@ -0,0 +1,278 @@
+    if os.path.exists(out_dir):
+        for item in os.listdir(out_dir):
+            if os.path.isdir(os.path.join(out_dir, item)):
+                output.append(item)
+    return output
+
+
+def get_lora_models(config: DreamboothConfig = None):
+    output = [""]
+    if config is None:
+        config = shared.db_model_config
+    if config is not None:
+        lora_dir = os.path.join(config.model_dir, "loras")
+        if os.path.exists(lora_dir):
+            files = os.listdir(lora_dir)
+            for file in files:
+                if os.path.isfile(os.path.join(lora_dir, file)):
+                    if ".pt" in file and "_txt.pt" not in file:
+                        output.append(file)
+    return output
+
+
+def get_sorted_lora_models(config: DreamboothConfig = None):
+    models = get_lora_models(config)
+
+    def get_iteration(name: str):
+        regex = re.compile(r'.*_(\d+)\.pt$')
+        match = regex.search(name)
+        return int(match.group(1)) if match else 0
+
+    return sorted(models, key=lambda x: get_iteration(x))
+
+
+def get_model_snapshots(config: DreamboothConfig = None):
+    snaps = [""]
+    if config is None:
+        config = shared.db_model_config
+    if config is not None:
+        snaps_dir = os.path.join(config.model_dir, "checkpoints")
+        if os.path.exists(snaps_dir):
+            for file in os.listdir(snaps_dir):
+                if os.path.isdir(os.path.join(snaps_dir, file)):
+                    rev_parts = file.split("-")
+                    if rev_parts[0] == "checkpoint" and len(rev_parts) == 2:
+                        snaps.append(rev_parts[1])
+    return snaps
+
+
+def unload_system_models():
+    try:
+        import modules.shared
+        if modules.shared.sd_model is not None:
+            modules.shared.sd_model.to("cpu")
+        for former in modules.shared.face_restorers:
+            try:
+                former.send_model_to("cpu")
+            except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```

Similar issue at line number 188.